packages/embla-carousel/src/CarouselTab.tsx

```diff
@@ -43,23 +42,19 @@ export const useCarouselTab = (
   const [selectedIndex, setSelectedIndex] = useState(0);
   const [scrollSnaps, setScrollSnaps] = useState<number[]>([]);
 
-  const handleClick = useCallback(
-    (index: number) => {
-      if (!emblaApi) return;
-      emblaApi.scrollTo(index);
-    },
-    [emblaApi],
-  );
+  const handleClick = (index: number) => {
+    emblaApi?.scrollTo(index);
+  };
 
-  const handleInit = useCallback((emblaApi: EmblaCarouselType) => {
-    setScrollSnaps(emblaApi.scrollSnapList());
-  }, []);
+  useEffect(() => {
+    const handleInit = (emblaApi: EmblaCarouselType) => {
```

Comment:
Moving handleInit and handleSelect inside the effect removes the need for useCallback